mne/decoding/time_gen.py

```diff
@@ -138,37 +143,53 @@ def fit(self, epochs, y=None):
         X, y, self.picks_ = _check_epochs_input(epochs, y, self.picks)
         self.ch_names = [epochs.ch_names[p] for p in self.picks_]
 
+        # Prepare cross-validation
         cv = self.cv
         if isinstance(cv, (int, np.int)):
-            cv = StratifiedKFold(y, cv)
-        cv = check_cv(cv, X, y, classifier=True)
-        self.cv_ = cv  # update CV
-        if not np.all([len(train) for train, _ in self.cv_]):
+            # Automatically chose StratifiedKFold if classification else KFold
+            try:
```

Comment:
There's only 3 trys in the code, I don't think that's excessive.

How else would you do this one? Like this?:

```
kargs = dict(cv=cv, y=y, classifier=is_classifier(self.clf))
if sklearn_new:
    args["X"] = X
cv = check_cv(**kargs)
```